### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by eywalker**
Initial migration of the notebook content into proper package/modules

### CODE REVIEW

File: probcore/distributions.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        ...
+
+
+class TrainableDistributionAdapter(nn.Module):
+    
+    n_rvs = 1
+    
+    def __init__(self, distribution_class, *dist_args, _parameters=None, **dist_kwargs):
+        super().__init__()
+        self.distrbrituion_class = distribution_class
```
Comment (suhasshrinivasan) [inline probcore/distributions.py]: correct spelling mistake: "self.distribution_class"

**Comment by suhasshrinivasan** [comment]
Everything looks good for just a typo which I commented on. If there are issues, they will become clear once the package starts getting used extensively for the neural-sampling-code project.

**Comment by eywalker** [comment]
Thanks @suhasshrinivasan! I fixed the spelling mistake. If the rest looks good, please accept the PR.

**Comment by suhasshrinivasan** [comment]
All good! Closing PR.

**Pull request merged**